Add option of coerce for EnumType

kaedub opened the pull request:
I have had to do some hacky workarounds to handle string case sensitivity when parsing an EnumType so I created this and was hoping to get it adopted. If this feature could cause problems in other places or if there is something I should change please let me know.
Thanks!

davidmdm commented:
@kaedub 
Hi! Sorry for not getting back to you quicker!

I see what you are trying to do and I think it's a fine change. If you can just add a little something in the readme documenting this addition to the EnumType API, I will merge it and release a new version on npm!

Thanks for your efforts again!

kaedub commented:
@davidmdm Thanks for the feedback. I went ahead and added to the Enum section of the readme. I noticed that I had not added the same logic to the Enum's check method so I added that in here too. I assume if we tell it to coerce strings to ignore casing, it should do so for both parse and check.

davidmdm commented:
No I think we should remove the logic from the check. I omitted a `check` method for most types because I wanted type coercion as a base feature of myzod, and I can't seem to make them play nice together.

If we leave the logic in you break the type system:

```typescript
enum RGB { red, green, blue }
const RGBSchema = myzod.enum(RGB).coerce('lower');

const maybeGreen = 'GREEN';
const g = RGBSchema.parse(maybeGreen);
// here g is of type RGB which is true because parse will have returned 'green'. 

if (RGBSchema.check(maybeGreen)) {
  // maybeGreen is typed as RGB here but the underlying value isn't and we've broken type-safety.
}
```

So I would remove it from the check, and make a note in the readme that coercion only affects the `parse` and `try` apis but not `check`.

Sounds good?

kaedub commented:
@davidmdm Yeah sounds good.

davidmdm commented on src/types.ts:
```diff
@@ -1198,33 +1198,46 @@ export class IntersectionType<T extends AnyType, K extends AnyType> extends Type
 
 type ValueOf<T> = T[keyof T];
+type EnumCoerceOptions = 'upper' | 'lower';
+export type EnumOptions<T> = {
```
could you add `coerce` and `default` methods to EnumType? I like enabling fluent syntax in the API

davidmdm commented on readme.md:
````diff
@@ -684,6 +684,24 @@ if (colorSchema.check(value)) {
+You can set `coerce` to `'lower'` or `'upper'` to ignore string casing when calling `check` or `parse`.
+
+```typescript
+z.enum(Colors, { coerce: 'lower' });
````
The example in the readme uses `check` but coerce specifically does not work with the `check` api

kaedub replied:
Ah thanks for catching this. Fix is up now. Might be best to squash this one at this point :D
